src/cipher.rs

```diff
@@ -64,35 +62,45 @@ pub extern "C" fn rustls_cipher_get_signature_scheme_name(
 ) {
     ffi_panic_boundary_unit! {
         let write_buf: &mut [u8] = unsafe {
-            let out_n: &mut size_t = match out_n.as_mut() {
-                Some(out_n) => out_n,
-                None => return,
-            };
-            *out_n = 0;
             if buf.is_null() {
                 return;
             }
             slice::from_raw_parts_mut(buf as *mut u8, len as usize)
         };
+        let out_n: &mut size_t = try_ref_from_ptr!(out_n, &mut size_t, ());
         let name = signature_scheme_name(scheme);
         let len: usize = min(write_buf.len() - 1, name.len());
         write_buf[..len].copy_from_slice(&name.as_bytes()[..len]);
-        unsafe {
-            *out_n = len;
-        }
+        *out_n = len;
     }
 }
 
-/// The complete chain of certificates plus private key for
-/// being certified against someones list of trust anchors (commonly
-/// called root store). Corresponds to `CertifiedKey` in the Rust API.
+/// The complete chain of certificates to send during a TLS handshake,
+/// plus a private key that matches the end-entity (leaf) certificate.
+/// Corresponds to `CertifiedKey` in the Rust API.
+/// https://docs.rs/rustls/0.19.0/rustls/sign/struct.CertifiedKey.html
 pub struct rustls_certified_key {
     // We use the opaque struct pattern to tell C about our types without
     // telling them what's inside.
     // https://doc.rust-lang.org/nomicon/ffi.html#representing-opaque-structs
     _private: [u8; 0],
 }
 
+/// Build a `rustls_certified_key` from a certificate chain and a private key.
+/// `cert_chain` must point to an array of `cert_chain_len` bytes, containing
```

Comment:
```suggestion
/// `cert_chain` must point to a buffer of `cert_chain_len` bytes, containing
```
Per discussion in Slack earlier today